tools/OMIcheck/OMIcheck.ps1

```diff
@@ -1,127 +1,460 @@
-﻿# This script lists the virtual machines in Azure subscriptions of the user,
+﻿# This script lists the virtual machines and scale sets in Azure subscriptions of the user,
 # and looks for vulnerable OMI packages. The report is grouped by subscription and resource group name.
 
 # Set below flag to true if you want vulnerable OMI to be patched.
-# If OMSLinuxAgent is installed, then a minor update will be triggered.
-# If OMI is standalone, the latest bits will be installed from Github.
-$upgradeOMI = $false; #detect only
-$omsVersionGood = "1.13"
+#    - If OMSLinuxAgent or LinuxDiagnostics is installed, then a minor update will be triggered.
+#    - OMI package is installed from Github.
+$upgradeOMI = $false #detect only
+
+$OmiServerGoodVersion = "OMI-1.6.8-1"
+$OmiPkgGoodVersion = "1.6.8.1"
+$LadPkgGoodVersion = "1.5.110-LADmaster.1483"
+$OmsPkgGoodVersion = "1.13.40.0"
+
+$LADpublisher = "Microsoft.Azure.Diagnostics"
+$LADextName = "LinuxDiagnostic"
+$OMSpublisher = "Microsoft.EnterpriseCloud.Monitoring"
+$OMSextName = "OmsAgentForLinux"
 
 # Update these paths accordingly. In cloud shell, the path is usually /home/user_name
 $checkScriptPath  = "$PSScriptRoot\omi_check.sh"
 $upgradeScriptPath = "$PSScriptRoot\omi_upgrade.sh"
 
-# Get all Azure Subscriptions
-Connect-AzAccount #-UseDeviceAuthentication
-$subs = Get-AzSubscription
+# Parse the response string.
+function ParsePkgVersions($str)
+{
+    $splitLines = $str.Split("`n",[System.StringSplitOptions]::RemoveEmptyEntries)
 
-# For each subscription execute to find (and upgrade) OMI on Linux VMs
-# TODO: This can be converted to ForEach-Object -Parallel. However, it requires PSh >= 7.1
-foreach ($sub in $subs)
+    $omiServerResp   = $splitLines | where { $_.Contains("/opt/omi/bin/omiserver:") }
```

Comment:
seems like we no longer do this check in the `.sh`, just `grep -E 'omi|omsagent|lad'`

Reply:
doing both  checks.

Reply:
Relying on the running server version made more sense.